packages/native-stack/src/views/HeaderConfig.tsx

```diff
@@ -182,6 +182,7 @@ export function HeaderConfig({
       backButtonInCustomView={backButtonInCustomView}
       backgroundColor={headerBackgroundColor}
       backTitle={headerBackTitleVisible ? headerBackTitle : ' '}
```

Comment:
Should we still consider `headerBackTitleVisible` in backTitle, if we're passing `backTitleVisible` below? I believe the button will be hidden anyway.

Reply:
I think it's necessary for people who are using older versions of `react-native-screens`, since `backTitleVisible` [is introduced in `react-native-screens` `3.21.0`](https://github.com/software-mansion/react-native-screens/commit/b5612bfefc27150b81d7f22c6b03a587acd44ac6#diff-28f909d0bd17e9a01ef7f575ba575e3726751335724532573627b503aa377508R42) and `@react-navigation/native-stack` has [peer dependency of `react-native-screens` as `>= 3.0.0`](https://github.com/react-navigation/react-navigation/blob/%40react-navigation/native-stack%407.0.0-alpha.20/packages/native-stack/package.json#L63).

(However, I haven't tested if only setting `backTitleVisible` will actually not work with `react-native-screens` `<3.21.0` for now.)

Reply:
I have now tested and can confirm that this is necessary for backward compatibility with `react-native-screens` versions `<3.21.0`. Without this, the back title will not be hidden while using `react-native-screens` `3.20.0`.

Comment added to explain this.

Reply:
That convinces me, thanks!